c10/core/impl/cow/state.cpp

```diff
@@ -0,0 +1,59 @@
+#include <c10/core/impl/cow/state.h>
+
+#include <c10/util/Exception.h>
+
+namespace c10::impl {
+
+auto cow::State::maybe_bump(cow::Simulator* maybe_simulator) -> void {
+  std::lock_guard<std::mutex> lock(mtx_);
+  if (!default_simulator_.has_value()) {
+    // Any created simulator should be bound to the storage that it
+    // was created from. Hence, there should only be an external
+    // simulator if the storage created it. We don't check for the
+    // specific matching of the storage and simulator, but we do check
+    // that the presence relationship holds.
+    //
+    // TODO: This check should probably be here, but it is actually
+    // triggering in the StaticRuntime.autogen_inner test.
+    //
+    // TORCH_INTERNAL_ASSERT(maybe_simulator == nullptr);
```

Comment:
It's being violated because you haven't handled what happens if you do `set_`.

Reply:
set_ that takes Storage is not possible to do precisely. set_ with Tensor can be done precisely.

Reply:
Resolution: wipe this to nullptr

Reply:
done for set with storage, not set with tensor yet. Set with tensor may be precise because we can just copy over the value of shadow_storage_ from the source tensor?